### PR TITLE
Remove fast_unwind_on_malloc and malloc_context_size restrictions in ASAN CI

### PR DESCRIPTION
**Pull request by tkf**
This implements @maleadt's suggestion https://github.com/JuliaLang/julia/issues/42498#issuecomment-934248231

A quick benchmark (https://github.com/JuliaLang/julia/issues/42498#issuecomment-935111691) suggests the peak memory consumption is about 5GB. I think the CI machines can handle this.

### CODE REVIEW

File: contrib/asan/Make.user.asan [hunk 1]
```diff
@@ -18,6 +18,3 @@ override WITH_GC_DEBUG_ENV=1
 
 # default to a debug build for better line number reporting
 override JULIA_BUILD_MODE=debug
-
-# make ASAN consume less memory
-export ASAN_OPTIONS=detect_leaks=0:fast_unwind_on_malloc=0:allow_user_segv_handler=1:malloc_context_size=2
```
[the remark this answers is not in the document]
Comment (tkf) [reply contrib/asan/Make.user.asan:23]: @maleadt added this in #41675

https://github.com/JuliaLang/julia/blob/5d87cb92a2abb5bc62dcd777af01a831de1f51cf/cli/loader_exe.c#L18-L20

**Review by maleadt — APPROVED** [review]
LGTM.

The options were mainly for reducing memory usage when running the tests, which apparently we do not do on CI (understandably so, it takes a really long time to finish the test suite under ASAN, but maybe it should perform some smoke tests at least?).

File: contrib/asan/Make.user.asan (hunk 1 above)
[the remark this answers is not in the document]
Comment (maleadt) [reply contrib/asan/Make.user.asan:23]: Yeah setting ASAN_OPTIONS overrides the embedded options, so this is good to test whether the embedded options still work (as they've been broken multiple times already).

**Comment by tkf** [comment]
There was a simple check to see if ASAN can detect a bug but we removed it #42229 since [my implementation `contrib/asan/check.jl`](https://github.com/JuliaLang/julia/blob/2f00fe1d10eb54ee697abf09169b396b9264cb53/contrib/asan/check.jl) was not reliable. But yes, it'd be nice to run some subset of tests. Maybe the ones that do some unsafe operations.

@DilumAluthge Do you want to wait for merging this until the issue in #41936 is resolved?

**Comment by DilumAluthge** [comment]
> Do you want to wait for merging this until the issue in #41936 is resolved?

Yeah I think that makes sense.

**Comment by maleadt** [comment]
Rebased.

**Comment by vtjnash** [comment]
We need to fix the ASAN tests first (https://github.com/JuliaLang/julia/issues/42540)

**Comment by maleadt** [comment]
Oh oops, I thought that had been fixed already. Sorry for the noise then.

**Pull request merged**